pkg/k8s/service_cache.go

```diff
@@ -107,20 +108,113 @@ type ServiceEvent struct {
 type ServiceNotification struct {
 	Action       CacheAction
 	ID           ServiceID
-	Service      *Service
-	OldService   *Service
-	Endpoints    *Endpoints
-	OldEndpoints *Endpoints
+	Service      *MinimalService
+	OldService   *MinimalService
+	Endpoints    *MinimalEndpoints
+	OldEndpoints *MinimalEndpoints
+}
+
+// MinimalService is a slimmed down version of 'Service'.
+// This serves as an intermediate step to switch over to the new load-balancer control-plane,
+// allowing implementation of an adapter without having to implement conversions of fields that
+// are unused.
+type MinimalService struct {
+	Labels      map[string]string
+	Annotations map[string]string
+	Selector    map[string]string
+}
+
+func (ms *MinimalService) DeepEqual(other *MinimalService) bool {
+	switch {
+	case ms == nil && other == nil:
+		return true
+	case ms == nil:
+		return false
+	case other == nil:
+		return false
+	default:
+		return maps.Equal(ms.Labels, other.Labels) &&
+			maps.Equal(ms.Annotations, other.Annotations) &&
+			maps.Equal(ms.Selector, other.Selector)
+	}
+}
+
+func (ms *MinimalService) IsExternal() bool {
+	return len(ms.Selector) == 0
+}
+
+func (ms *MinimalService) DeepCopy() *MinimalService {
+	return &MinimalService{
+		Labels:      maps.Clone(ms.Labels),
+		Annotations: maps.Clone(ms.Annotations),
+		Selector:    maps.Clone(ms.Selector),
+	}
+}
+
+func newMinimalService(svc *Service) *MinimalService {
+	if svc == nil {
+		return nil
+	}
+	return &MinimalService{}
+}
+
+// MinimalService is a slimmed down version of 'Endpoints'.
+// This serves as an intermediate step to switch over to the new load-balancer control-plane,
+// allowing implementation of an adapter without having to implement conversions of fields that
+// are unused.
+type MinimalEndpoints struct {
+	Backends map[cmtypes.AddrCluster]serviceStore.PortConfiguration
 }
 
-// ServiceCache is a list of services correlated with the matching endpoints.
+func (meps *MinimalEndpoints) Prefixes() []netip.Prefix {
+	prefixes := make([]netip.Prefix, 0, len(meps.Backends))
+	for addrCluster := range meps.Backends {
+		addr := addrCluster.Addr()
+		prefixes = append(prefixes, netip.PrefixFrom(addr, addr.BitLen()))
+	}
+	return prefixes
+}
+
+func newMinimalEndpoints(eps *Endpoints) *MinimalEndpoints {
+	if eps == nil {
+		return nil
+	}
+	meps := &MinimalEndpoints{
+		Backends: map[cmtypes.AddrCluster]serviceStore.PortConfiguration{},
+	}
+	for addrCluster, cfg := range eps.Backends {
+		meps.Backends[addrCluster] = cfg.Ports
+	}
+	return meps
+}
+
+type ServiceCache interface {
+	Events() <-chan ServiceEvent
```

Comment:
suggestion: Events() is not really for general use; only one consumer is allowed. We should call this out in the docblock, and maybe even `Error("BUG...")` if called more than once for the same ServiceCache.

Reply:
hmm, tests call it multiple times. OK if we'll just document this? It's not going to live very long. And in the operator-side use of this we can make that one unexported.

Reply:
deal.

Reply:
And yes, it's not a good API :-).